lib/utils.pm

```diff
@@ -753,17 +753,19 @@ sub handle_login {
         }
         type_string "root\n";
     }
-    if (get_var('DM_NEEDS_USERNAME') and !get_var('ROOTONLY')) {
-        type_string "$username\n";
-    }
-    if (check_var('DESKTOP', 'gnome') || (check_var('DESKTOP', 'lxde') && check_var('VERSION', '42.1'))) {
-        # DMs in condition above have to select user
-        if ((is_sle && sle_version_at_least('15')) || (is_leap && leap_version_at_least('15.0')) || is_tumbleweed) {
-            assert_and_click "displaymanager-$username";
-            record_soft_failure 'bgo#657996 - user account not selected by default, have to use mouse to login';
+    else {
+        if (get_var('DM_NEEDS_USERNAME')) {
```

Comment:
I don't think the encapsulated additional if within else is more readable then `elsif` to not introduce another indendation level ->

```perl
elsif (get_var('DM_NEEDS_USERNAME')) {
…
}
elsif (check_var('DESKTOP', 'gnome')) {
…
}
else {
    send_key 'ret';
}
```

or am I reading something wrong?

Reply:
That was my other approach, but felt this way was more readable. Will change it to elsif blocks as suggested.